gammapy/stats/feldman_cousins.py

```diff
@@ -0,0 +1,436 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Provide a Feldman Cousins algorithm."""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import numpy as np
+import matplotlib.pylab as plt
+from scipy import stats
+
+__all__ = ['find_confidence_interval_gauss',
+           'find_confidence_interval_poisson'
+           'construct_confidence_belt_PDFs',
+           'get_upper_and_lower_limit',
+           'fix_upper_and_lower_limit',
+           'find_limit',
+           'find_average_upper_limit',
+           'construct_confidence_belt']
```

Comment:
I added two example files that produce Fig. 7 and 10 from the Feldman Cousins paper. Due to a matplotlib problem, I cannot test them currently.

The API is definitely only expert friendly. The main feature is the function fc_construct_confidence_belt_pdfs that implements Feldman Cousins in a numerical way. This is useful if your statistic is not Poisson or a Gaussian bounded at the origin. I implemented fc_find_confidence_interval_poisson and fc_find_confidence_interval_gauss to show that my numerical solution works.

Long term it would be nice to wrap these functions into classes (e.g. like TFeldmanCousins in ROOT is a wrapper around the functionality of fc_find_confidence_interval_gauss).